### DISCUSSION TITLE
How best to do conditional routing?

### DISCUSSION THREAD

**Discussion by potch** [remix-run/remix#6978] [Q&A] [answered]
I have a site with routes like this:

```
thing_.$slug
thing_.$slug.overview
thing_.$slug.details
thing_.$slug.extra
```

if visitors land on `thing_.$slug`, I'd like to redirect them to `thing_.$slug.overview`. `thing_.$slug` *does* render UI.

What's the best way to do this sort of thing? My first thought was checking router matches in the loader/action then firing a conditional redirect if it's the "bottom-most" route, but I'm not sure how to get route info in loaders.

**Comment by kiliman** [comment, marked as answer]
You could add a route: `thing_.$slug._index.tsx` and create a loader that simply redirects to overview.

```ts
// thing_.$slug._index.tsx
import { redirect } from '@remix-run/node'

export const loader = () => redirect('overview')
```

**Reply by potch** [reply]
I completely spaced on the index routes- that's a perfect solution for this use case. Thank you! I'm still hopeful I could do some route introspection in loaders/actions but this works great.
